proto/OptimizerConfig.proto

```diff
@@ -102,22 +104,17 @@ message AdadeltaOptimizerState {
   optional TensorProto update_delta = 4;
 }
 
+
 message AdagradOptimizerState {
-  // learning rate policy
-  optional double learning_rate = 101;
-  optional double lr_decay_a = 102;
-  optional double lr_decay_b = 103;
+  optional LrPolicyState lr_state = 101;
```

Comment:
Once the protobuf index is used by messages, it should never change for compatibility, for the old client or somewhere is still using the old `.proto` generated code.

1. Do not change the current index
1. Only append field not modify them.